pymc/sampling/mcmc.py

```diff
@@ -1369,12 +1372,17 @@ def _init_jitter(
     if not jitter:
         return [ipfn(seed) for ipfn, seed in zip(ipfns, seeds)]
 
-    model_logp_fn: Callable
+    model_logp_fn: Callable[[PointType], np.ndarray]
     if logp_dlogp_func is None:
-        model_logp_fn = model.compile_logp()
+        if logp_fn is None:
+            # pymc NUTS path
+            model_logp_fn = model.compile_logp()
+        else:
+            # Jax path
+            model_logp_fn = logp_fn
     else:
 
-        def model_logp_fn(ip):
+        def model_logp_fn(ip: PointType) -> np.ndarray:
             q, _ = DictToArrayBijection.map(ip)
             return logp_dlogp_func([q], extra_vars={})[0]
```

Comment:
Let's simplify this. And make `init_nuts` pass a logp function (by wrapping the `logp_dlogp_func` in that case, similarly to how jax is wrapping it's logp function.

Then there is just a `if logp_func is None` or not inside

Reply:
Have changed to incorporate your feedback